Gutenberg/version.rb

```diff
@@ -11,8 +11,13 @@
 #
 #   LOCAL_GUTENBERG=../my-gutenberg-fork bundle exec pod install
 GUTENBERG_CONFIG = {
-  # commit: '',
-  tag: 'v1.97.1'
+  # To facilitate the adoption of the XCFramework setup, which greatly speeds up `pod install`,
+  # we shipped an ad hoc commit build with the v1.97.1 code,
+  # even though the codebase for that tag did not have the necessary automation.
+  # This approach will no longer be necessary starting v1.98.0,
+  # which should be cut from `trunk` at a point where all the necessary automation is already in place.
+  commit: 'ce86d9eafeb6b83c38eb7f4745d4b9be96675778'
+  # tag: 'v1.97.1'
```

Comment:
I'm wondering if we should manually build and upload xcframeworks for `v1.97.1` (maybe even including a few releases back too). For two reasons: it's not obvious what release this commit belongs and more importantly someone may want to use an older releases for whatever reason (like verifying a bug), but they won't be able to do that because the old release doesn't exist in the S3 bucket.

Reply:
> I'm wondering if we should manually build and upload xcframeworks for v1.97.1 (maybe even including a few releases back too

This can be done. Sure 👍 

However, the build for v1.98.0 is available, so there is no blocking concern to merging this. If that's okay with you, I'll rebase on `trunk` (and loose the "changes since you last saw this") so that we can use that build. Otherwise, I can put make a merge commit.

Reply:
I'm okay with either way. Whatever is convenient for you would work for me 😃

Reply:
I rebased. 

Unfortunately, as CI will show, since be bumped the iOS deployment target from 14 to 15, Gutenberg 1.98.0 no longer works:

<img width="1277" alt="image" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/9041d5e8-0499-491f-84e7-6eca1e8b6493">

So, we're back to square one in regards to having to manually publish new versions. 

In parallel, there's work going on updating React Native on the Gutenberg end, which may be best suited as the version where to make the switch 😞 

- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5908
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5912

Reply:
> In parallel, there's work going on updating React Native on the Gutenberg end, which may be best suited as the version where to make the switch 😞
> 
> * [Workaround RNReanimated `pod install` failure gutenberg-mobile#5908](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5908)
> * [Workaround React Native 0.71 and XCFramework setup issues (without Hermes) gutenberg-](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5912)

Since the RN upgrade is blocked due to issues with the Reanimated Pod, which would be solved using the XCFramework. Maybe we could link this work with the ongoing work of the upgrade. I.e. the RN upgrade and the XCFramwork integration would land at the same time.

We could wrap up the workaround ([Workaround React Native 0.71 and XCFramework setup issues (without Hermes)](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5912/files#top)) and merge it into the [Gutenberg Mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5881). And then, we could rebase this branch with the changes needed for the upgrade in WP-iOS ([PR reference](https://github.com/wordpress-mobile/WordPress-iOS/pull/20956)), including updating the Gutenberg Mobile reference. WDYT?

Reply:
Sounds good to me 👍

Reply:
@mokagio I'm trying to test the workaround for solving the issue with Reanimated, but when installing the pods in WP-iOS I'm getting the error shared [here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5908).

```
Error: Cannot find module 'react-native/package.json'
Require stack:
- /Users/fluiddot/workspace/WordPress-iOS/[eval]
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:931:15)
    at Function.resolve (internal/modules/cjs/helpers.js:113:19)
    at [eval]:1:9
    at Script.runInThisContext (vm.js:134:12)
    at Object.runInThisContext (vm.js:310:38)
    at internal/process/execution.js:81:19
    at [eval]-wrapper:6:22
    at evalScript (internal/process/execution.js:80:60)
    at internal/main/eval_string.js:27:3 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/Users/fluiddot/workspace/WordPress-iOS/[eval]' ]
}

[!] Invalid `RNReanimated.podspec` file: no implicit conversion of nil into String.

 #  from /var/folders/s7/lwl9q12j2_31hw8y1vkmln180000gn/T/d20230630-7085-146eqk6/RNReanimated.podspec:5
 #  -------------------------------------------
 #  reanimated_package_json = JSON.parse(File.read(File.join(__dir__, "package.json")))
 >  config = find_config()
 #  assert_no_multiple_instances(config)
 #  -------------------------------------------
```

I'm wondering if we'd need to specify as done [here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5912/files#diff-141dafd8df7f3d127b6295567b663372841c0bc9f142fcf433b6858cc3f95cf3R57) that Reanimated should be fetched from the forked repository. WDYT?

On a similar note, I wanted to ask if now, that we integrate Gutenberg using the XCFramework, we still need to reference and install the Pod dependencies of Gutenberg.

https://github.com/wordpress-mobile/WordPress-iOS/blob/0f8cc000b479d2ad10fea632155c186f4c597ee1/Gutenberg/cocoapods_helpers.rb#L14-L60

Thanks you very much for the help on this 🙇 !

Reply:
Hey @fluiddot 👋 

Is the error occurring on this branch? A standard `bundle exec pod install` should not result in that error ([as can be seen in CI](https://buildkite.com/automattic/wordpress-ios/builds/15153#01891a4f-cc58-4d21-8967-1e6d4d2b0bed)) because of the XCFramework.

Reply:
> Is the error occurring on this branch? A standard `bundle exec pod install` should not result in that error ([as can be seen in CI](https://buildkite.com/automattic/wordpress-ios/builds/15153#01891a4f-cc58-4d21-8967-1e6d4d2b0bed)) because of the XCFramework.
>
> The only way I can think for that error happening is when running using a local version of Gutenberg, `LOCAL_GUTENBERG=true bundle exec pod install`?

@mokagio Yes, but as you pointed out, I encountered this error when using the local version of Gutenberg. It works If I run `bundle exec pod install` using the tag `v1.98.0` as the Gutenberg Mobile reference 👍 .

> I'll take a look at that flow now. FWIW, that seems consistent with what @geriux [reported](https://github.com/wordpress-mobile/WordPress-iOS/pull/20717#issuecomment-1603142170) during earlier testing.

That would be great, thanks 🙇 ! We rely on using the local version of Gutenberg when testing and debugging changes we need to make on the native side related to the editor. Hence, being able to keep using it would be very helpful for the project. Let me know if I can help anyhow on this part.